doc/developer/projects.rst

```diff
@@ -0,0 +1,77 @@
+Mentored Projects
+==================
+
+This page maintains a list of mentored project ideas that contributors can work
+on if they are interested in contributing to the NetworkX project. Feel free to
+suggest any other idea if you are interested on the
+`NetworkX GitHub discussions page <https://github.com/networkx/networkx/discussions>`__ 
+
+These ideas can be used as projects for Google Summer of Code, Outreachy,
+NumFOCUS Small Development Grants and university course/project credits (if
+your university allows contribution to open source for credit).
```

Comment:
A bit more background here would be good: something about how these projects are expected to be larger in scope than a single PR, i.e. appropriate for: <your list, GSoc, semester projects, etc.>.